ToastTests/Foundation/Block Enumeration/TWTBlockEnumerationTests.m

```diff
@@ -237,6 +257,27 @@ - (void)testCollectionBlockEnumerationCollect
 }
 
 
+- (void)testCollectionBlockEnumerationFlatten
+{
+    NSMutableArray *randomArray = [[NSMutableArray alloc] init];
+
+    for (NSUInteger i = 0; i < [self randomCount]; ++i) {
+        [randomArray addObject:[self randomStringArray]];
+    }
+    
+    NSMutableArray *expectedArray = [[NSMutableArray alloc] init];
+    for (NSArray *randomStringArray in randomArray) {
+        [expectedArray addObjectsFromArray:randomStringArray];
+    }
+
+    NSArray *flattenedArray = [randomArray twt_flatten];
+    
+    XCTAssertNotNil(flattenedArray, @"Returned collection is nil");
+    XCTAssertEqual([expectedArray count], [flattenedArray count], @"Returned collection does not match size of original collection");
+    XCTAssertEqualObjects(expectedArray, flattenedArray, @"Flattened collection does not match the expected collection");
+}
```

Comment:
Need tests on each of the receiving collection classes. I think trying this on anything other than `NSArray` will expose the bug I mentioned above.